データベース設計

mkajitani15 opened the pull request:
# What
データベース設計をReadmeに記述する。

# Why
chat-space開発に伴い、データベース設計を行う必要があるため。

mkajitani15 committed:
create pull request

mkajitani15 committed:
edit README_DatabaseDesign

READMEにデータベース設計を記載。

40213151 commented on README.md:
```diff
@@ -22,3 +22,48 @@ Things you may want to cover:
+## usersテーブル
+|Column|Type|Options|
+|------|----|-------|
+|name|text|null: false, foreign_key: true|
```
emaiとnameはstring型にしましょう！

40213151 commented on README.md:
```diff
@@ -22,3 +22,48 @@ Things you may want to cover:
+## groupsテーブル
+|Column|Type|Options|
+|------|----|-------|
+|name|text|null: false, foreign_key: true|
```
nameはstring型にしましょう！

40213151 commented on README.md:
```diff
@@ -22,3 +22,48 @@ Things you may want to cover:
+
+
+## commentsテーブル
+|Column|Type|Options|
```
chatspaceではメッセージに画像を添付する機能があります
なので画像カラムを付け足しましょう！

40213151 commented on README.md:
```diff
@@ -22,3 +22,48 @@ Things you may want to cover:
+## commentsテーブル
+|Column|Type|Options|
+|------|----|-------|
+|comment|text|null: false, foreign_key: true|
```
こちらもstring型に修正しましょう！

40213151 reviewed (commented):
コメントしましたのでご確認お願いします！

mkajitani15 committed:
Update Readme_DatabaseDesign

各テーブルのデータ型(type)の修正。
コメントテーブルへのimage絡むの追加

mkajitani15 commented:
修正実施しました。ご確認お願いしますmm
- users：emaiとnameはstring型
- groups：nameはstring型
- comments：commentsをstring型、imageカラムの追加

shingoteshima commented on README.md:
```diff
@@ -22,3 +22,49 @@ Things you may want to cover:
+|Column|Type|Options|
+|------|----|-------|
+|comment|string|null: false, foreign_key: true|
+|image|string|null: false, foreign_key: true|
```
テキスト、写真は単体で投稿可能なのでnull: falseは不要です。
また、foreign_key: trueも不要です。

mkajitani15 committed:
Update Readme

コメントテーブル
commentカラムおよびimageカラムからnull: false, foreign_key: trueを削除。

mkajitani15 commented:
修正実施しました。
- comments：commentカラムおよびimageカラムから `null: false, foreign_key: true` を削除。

i-date commented on README.md:
```diff
@@ -22,3 +22,49 @@ Things you may want to cover:
+## usersテーブル
+|Column|Type|Options|
+|------|----|-------|
+|name|string|null: false, foreign_key: true|
```
`foreign_key: true`は外部キー制約なので、nameカラムには不要ですので削除しましょう。
外部キーが分からないようでしたら調べてみましょう。

i-date commented on README.md:
```diff
@@ -22,3 +22,49 @@ Things you may want to cover:
+|Column|Type|Options|
+|------|----|-------|
+|name|string|null: false, foreign_key: true|
+|email|string|null: false, foreign_key: true|
```
emailカラムはdeviseで自動生成されるため、不要ですので削除して構いません。

i-date commented on README.md:
```diff
@@ -22,3 +22,49 @@ Things you may want to cover:
+## groupsテーブル
+|Column|Type|Options|
+|------|----|-------|
+|name|string|null: false, foreign_key: true|
```
groupsテーブルのnameカラムも`foreign_key: true`を削除しましょう。

i-date commented on README.md:
```diff
@@ -22,3 +22,49 @@ Things you may want to cover:
+## commentsテーブル
+|Column|Type|Options|
+|------|----|-------|
+|comment|string||
```
commentカラムは長文投稿も考え、text型にしましょう。

i-date commented on README.md:
```diff
@@ -22,3 +22,49 @@ Things you may want to cover:
+|comment|string||
+|image|string||
+|user_id|integer|null: false, foreign_key: true|
+|group_id|integer|null: false, foreign_key: true|
```
53-54行目の外部キーはinteger型ではなく、references型を使用しましょう。
そうでない場合、`foreign_key: true`では外部キー制約にならないので注意です。
また、そうすることで `user_id` を `user` と記述するとテーブルが生成される時に _id が付与されます。
例：`|user|references|null: false, foreign_key: true|`
あと他の外部キーも修正しましょう。
参考：https://qiita.com/ryouzi/items/2682e7e8a86fd2b1ae47

i-date commented on README.md:
```diff
@@ -22,3 +22,49 @@ Things you may want to cover:
+|Column|Type|Options|
+|------|----|-------|
+|user_id|integer|null: false, foreign_key: true|
+|group_id|integer|null: false, foreign_key: true|
```
65-66行目もinteger型ではなく、references型を使用しましょう。

i-date reviewed (commented):
追加コメントいたしました^^

mkajitani15 commented:
memo
- [x] 外部キーについて理解する。
- [x] foreign_key: trueは外部キー制約なので、nameカラムには不要ですので削除しましょう。
- [x] emailカラムはdeviseで自動生成されるため、不要ですので削除して構いません。
- [x] groupsテーブルのnameカラムもforeign_key: trueを削除しましょう。
- [x] commentカラムは長文投稿も考え、text型にしましょう。
- [x] 53-54行目の外部キーはinteger型ではなく、references型を使用しましょう。
- [x] そうでない場合、foreign_key: trueでは外部キー制約にならないので注意です。
- [x] そうすることで user_id を user と記述するとテーブルが生成される時に _id が付与されます。
例：|user|references|null: false, foreign_key: true|
- [x] あと他の外部キーも修正しましょう。
参考：https://qiita.com/ryouzi/items/2682e7e8a86fd2b1ae47
- [x] 65-66行目もinteger型ではなく、references型を使用しましょう。

mkajitani15 committed:
Update Readme

usersテーブル:emailカラム削除
usersテーブルおよびgroupsテーブル:外部キー制約削除
commentsテーブルおよびgroups_usersテーブル:references型に変更

mkajitani15 commented:
修正しました！ご確認お願いしますmm

- usersテーブル,groupsテーブル：外部キー制約削除
- usersテーブル：emailカラム削除
- commentsテーブル：commentカラムをtext型に変更
- commentsテーブル,groups_usersテーブル： 外部キーの箇所をreferences型に変更、カラム名変更

shingoteshima commented:
LGTMです！
